tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py

```diff
@@ -0,0 +1,2164 @@
+        self, dev, diff_method, grad_on_execution, max_diff
+    ):
+        """Test that the Hamiltonian is expanded if there
+        are non-commuting groups and the number of shots is finite
+        and the first and second order gradients are correctly evaluated"""
+        gradient_kwargs = {}
+        tol = 0.1
+        if diff_method in ("adjoint", "backprop"):
+            pytest.skip("The adjoint and backprop methods do not yet support sampling")
+        elif diff_method == "spsa":
+            gradient_kwargs = {"h": H_FOR_SPSA}
+            np.random.seed(SEED_FOR_SPSA)
+            tol = TOL_FOR_SPSA
+        elif diff_method == "finite-diff":
+            gradient_kwargs = {"h": 0.05}
+            tol = 0.15
+        elif diff_method == "hadamard":
+            pytest.skip("The hadamard method does not yet support Hamiltonians")
+
+        obs = [qml.PauliX(0), qml.PauliX(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliZ(1)]
+
+        @qnode(
+            dev,
+            diff_method=diff_method,
+            grad_on_execution=grad_on_execution,
+            max_diff=max_diff,
+            interface="torch",
+            **gradient_kwargs,
+        )
+        def circuit(data, weights, coeffs):
+            weights = torch.reshape(weights, [1, -1])
+            qml.templates.AngleEmbedding(data, wires=[0, 1])
+            qml.templates.BasicEntanglerLayers(weights, wires=[0, 1])
+            H = qml.Hamiltonian(coeffs, obs)
+            return qml.expval(H)
+
+        d = torch.tensor([0.1, 0.2], requires_grad=False, dtype=torch.float64)
+        w = torch.tensor([0.654, -0.734], requires_grad=True, dtype=torch.float64)
+        c = torch.tensor([-0.6543, 0.24, 0.54], requires_grad=True, dtype=torch.float64)
+
+        # test output
+        res = circuit(d, w, c, shots=50000)
+
+        expected = c[2] * torch.cos(d[1] + w[1]) - c[1] * torch.sin(d[0] + w[0]) * torch.sin(
+            d[1] + w[1]
+        )
+        assert torch.allclose(res, expected, atol=tol)
+
+        # test gradients
+        if diff_method in ["finite-diff", "spsa"]:
+            pytest.skip(f"{diff_method} not compatible")
+
+        res.backward()
+        grad = (w.grad, c.grad)
+
+        expected_w = torch.tensor(
+            [
+                -c[1] * torch.cos(d[0] + w[0]) * torch.sin(d[1] + w[1]),
+                -c[1] * torch.cos(d[1] + w[1]) * torch.sin(d[0] + w[0])
+                - c[2] * torch.sin(d[1] + w[1]),
+            ]
+        )
+        expected_c = torch.tensor(
+            [0, -torch.sin(d[0] + w[0]) * torch.sin(d[1] + w[1]), torch.cos(d[1] + w[1])]
+        )
+
+        assert torch.allclose(grad[0], expected_w, atol=tol)
+        assert torch.allclose(grad[1], expected_c, atol=tol)
+
+        # test second-order derivatives
+        if diff_method == "parameter-shift" and max_diff == 2:
+            hessians = torch.autograd.functional.hessian(
+                lambda _d, _w, _c: circuit(_d, _w, _c, shots=50000), (d, w, c)
```

Comment:
No need to change this but `functools.partial` could be used here as well (I generally prefer it over lambdas)

Reply:
guess I got used to lambdas, but it's actually what [torch recommends](https://pytorch.org/docs/stable/autograd.html#functional-higher-level-api) 😄

Reply:
😮